fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -603,10 +625,9 @@ def _validate_input(self, X, y, A, reinitialize=False):
                 dtype=float,
                 allow_nd=True,
             )
-            y = self._validate_data(y, ensure_2d=False)
+            y = self._validate_data(y, dtype=None, ensure_2d=False)
```

Comment:
note for a separate PR, sklearn>=1.6 now has `sklearn.utils.validation.validate_data` instead. So we need to move this into the `fixes.py` and use that when available.